diff --git a/src/drivers/display/drm/lv_linux_drm_egl.c b/src/drivers/display/drm/lv_linux_drm_egl.c
index 546256611..66e665d32 100644
--- a/src/drivers/display/drm/lv_linux_drm_egl.c
+++ b/src/drivers/display/drm/lv_linux_drm_egl.c
@@ -757,4 +757,12 @@ static void drm_destroy_window(void * driver_data, void * native_window)
 }
 
 
+int lv_linux_drm_get_fd(lv_display_t * disp)
+{
+    if(!disp) return -1;
+    lv_drm_ctx_t * ctx = lv_display_get_driver_data(disp);
+    if(!ctx) return -1;
+    return ctx->fd;
+}
+
 #endif /*LV_USE_LINUX_DRM && LV_LINUX_DRM_USE_EGL*/
